SSA/Projects/DCE/DCE.lean

```diff
@@ -0,0 +1,492 @@
+import SSA.Core.Framework
+
+/-- Delete a variable from a list. -/
+def Ctxt.delete (Γ : Ctxt Ty) (v : Γ.Var α) : Ctxt Ty :=
+  Γ.removeNth v.val
+
+/-- Witness that Γ' is Γ without v -/
+def Deleted {α : Ty} (Γ: Ctxt Ty) (v : Γ.Var α) (Γ' : Ctxt Ty) : Prop :=
```

Comment:
`Deleted` should be in a namespace, say `Ctxt.Deleted`

Reply:
I propose we refactor this in the next commit. I agree that this ought to be moved.